From 7de968a470ee66c2ce8d0975a4945c742091334b Mon Sep 17 00:00:00 2001
From: Zhi Jin <zhi.jin@intel.com>
Date: Wed, 28 Nov 2012 14:25:53 +0800
Subject: [PORT FROM MAIN]audio: check the state of RTSPSource before stop it

BZ: 70058

Stopping a uninitialized RTSPSource will trigger tombstone, so
check the state of it before stopping it.

Change-Id: If0dc5cddd39c5e6a1a48e280bbb23d62bbe8778f
Orig-Change-Id: Ia97831f99b846054d98a991d0babfaed3fb26165
Signed-off-by: Zhi Jin <zhi.jin@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 .../libmediaplayerservice/nuplayer/RTSPSource.cpp  |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
index 51075ec..12fea3b 100644
--- a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
+++ b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
@@ -115,6 +115,11 @@ void NuPlayer::RTSPSource::stop() {
     if (mLooper == NULL) {
         return;
     }
+    if (mState == DISCONNECTED) {
+        ALOGI("already disconnected.");
+        return;
+    }
+
     sp<AMessage> msg = new AMessage(kWhatDisconnect, mReflector->id());
 
     sp<AMessage> dummy;
-- 
1.7.4.1

